lib/segment/src/vector_storage/query_scorer/mod.rs

```diff
@@ -29,7 +30,8 @@ pub fn score_max_similarity<TMetric: Metric<VectorElementType>>(
         sum += multi_dense_b
             .iter()
             .map(|dense_b| TMetric::similarity(dense_a, dense_b))
-            .fold(ScoreType::NEG_INFINITY, |a, b| if a > b { a } else { b });
+            .max_by(|a, b| OrderedFloat(*a).cmp(&OrderedFloat(*b)))
```

Comment:
I replaced the `fold` by a more explicit implementation using `max_by`

Also used `OrderedFloat` for consistency with our `ScoredPoint` implementation.

Reply:
Thanks! Looks much better